### ISSUE TITLE
Multiple Container Registries Login Issue

### ISSUE THREAD

**Issue by elliotmoule** [Microsoft/vscode-azure-iot-edge#260]
I noticed that if I use a second azure container registry, I am required to manually login into the ACR:

![image](https://user-images.githubusercontent.com/29968849/46057919-8a594d00-c19b-11e8-8aa5-c8bdc1e7f6c3.png)

It appears to stick for some time, as I'm able to continue pushing for the next while, however, from what I can tell, it does expire, whereas the other container registry I use (the first one) has never required me to do this. I was kind of assuming it was pulling the login information from the .env file, but I'm not sure now?

No rush to fix, as manual log in works, but just pointing it out. 

Thanks!

Elliot Moule

**Comment by LazarusX** [comment]
Thanks for your kind feedback!

A couple of follow-up questions though:
* For your second ACR, was there any change to the credential somewhere else when it expired? 
* Did that happen only for a specific ACR?
* Did you login with Azure AD, or ACR admin password, or service principle (see the difference [here](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication))? If you login with Azure AD, there is indeed an [expiration mechanism](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication#individual-login-with-azure-ad) which I suppose is out of our control.

Currently, we won't do `docker login` or `az acr login` for you with the information in the .env file. It is only used to fill the registry credential section in the deployment manifest. However, we will definitely consider improving the login experience.

**Comment by elliotmoule** [comment]
Hi @LazarusX ,

Apologies for the slow reply, I didn't get a notification, though I should have checked sooner.

Addressing your questions:

- No change to the credential, has been the same since creating the service.
- It's hard to tell, as I only have two, with the second being the one I had trouble with. I haven't gone as far as creating a third to test.
- I have been logged in with my AzureAD, though, my first ACR was still working, kind of indicating that the account/token was still valid? Though, when I logged in (in the above example), I used the ACR admin password, so perhaps through having done that, and thus having two mechanisms active, it may have invalidated the second? 

As a side note, the issue hasn't happened again, since I originally posted, so not sure the difference, but I did follow the same proceedure as the above (if any others encounter the same issue).

Thanks for all your help, hopefully I've been helpful (doesn't feel like it!).

Regards,

Elliot Moule.
